src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Using 'Executors.newCachedThreadPool()' can lead to performance issues if a large number of threads are created. Consider using a ThreadPoolExecutor with a set maximum pool size and a keep-alive time to better control resource usage.